### PR TITLE
Add ability to forward incident to an external party 

### PR DESCRIPTION
**Pull request by MauritsR**
Part of https://github.com/Signalen/product-steering/issues/286. 

### Context

This PR adds a new form to forward incidents to an external party, for them to pick up the issue. 

When the feature flag is turned on, users are able to press the "Extern" button in the incident detail page. A form is displayed for e-mail, a text field and the images (if available) that will be visible to the external party. When submitted, an e-mail template is shown which can be confirmed. On confirmation, the incident status is changed (Doorgezet naar extern) and the e-mail is sent to the external party. 

The second part of the flow (reply from the external party) will be submitted in a separate PR. 

### Changes

- Adds `enableForwardIncidentToExternal` feature flag
- Adds `ForwardToExternal` form
- Create ModalDialog component used by EmailPreview and DefaultTexts
- Adds `DOORGEZET_NAAR_EXTERN` status
- Wrap buttons on incident detail page to prevent horizontal scroll bar on mobile

## Design

InVision designs can be found [here](https://projects.invisionapp.com/share/RF131I9Q6DVM#/screens/470029477)

## Testing this branch

- Set `enableForwardIncidentToExternal` to `true` in your json config (e.g. `app.json`)
- Run the backend using the [appropriate branch](https://github.com/Signalen/backend/tree/product-steering/261-prototype), as seen in PR #https://github.com/Amsterdam/signals/pull/1106. 
- In the [Django admin](http://localhost:8000/signals/admin/email_integrations/emailtemplate/), add the below email template for the key `Send mail signal forwarded to external`. This example corresponds to the design but any municipality can choose their own template. 

```
Geachte behandelaar,

Er is een melding binnengekomen bij de Gemeente. Kunnen jullie hier naar kijken en ons laten weten of jullie deze kunnen afhandelen?

**Wat kan u doen?**\
U kan de melding gelijk inzien en oppakken. Als de melding is verwerkt ontvangen wij graag een bericht.\
[Bekijk de actie]({{ reaction_url }})

Nummer: {{ formatted_signal_id }}

Met vriendelijke groet,

{{ ORGANIZATION_NAME }}
```

### CODE REVIEW

File: src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+          <TextArea
+            id="externalEmailMessage"
+            maxContentLength={MAX_MESSAGE_LENGTH}
+            rows={12}
+            {...register('message')}
+          />
+        </ErrorWrapper>
+      </StyledSection>
+
+      {attachments && attachments?.count > 0 && (
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx:185]: `attachments &&` can be removed.

Comment (vdegraaf) [inline src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx:185]: `attachments &&` can be removed.

**Review by vdegraaf — APPROVED** [review]
Works and looks really clean! Just had one comment on the code and: 
- the updated status in the notes is different then you choose. Is that intentional?

Screenshot 2022-12-12 at 13 18 06

File: src/signals/incident-management/containers/IncidentDetail/components/EmailPreview/EmailPreview.tsx [hunk 2]
```diff
@@ -30,9 +23,11 @@ const StyledIframe = styled.iframe`
 `
 
 interface EmailPreviewProps {
-  emailBody: string
+  emailBody?: string
```
Comment (OscarBakker) [inline src/signals/incident-management/containers/IncidentDetail/components/EmailPreview/EmailPreview.tsx:26]: maybe the optionals should be last?

File: src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  StyledH2,
+  StyledParagraph,
+  StyledSection,
+} from './styled'
+
+export const MAX_MESSAGE_LENGTH = 3000
+const schema = yup.object({
+  email: yup
+    .string()
+    .email('Dit is geen geldig e-mail adres')
```
Comment (OscarBakker) [inline src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx:39]: this error message doesnt have a DOT

File: src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+          <TextArea
+            id="externalEmailMessage"
+            maxContentLength={MAX_MESSAGE_LENGTH}
+            rows={12}
+            {...register('message')}
+          />
+        </ErrorWrapper>
+      </StyledSection>
+
+      {attachments && attachments.count > 0 && (
```
Comment (OscarBakker) [inline src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx:185]: can be attachments?.count right?

**Review by OscarBakker — APPROVED** [review]
nice contribution! some comments

File: src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx (hunk 4 above)
Comment (OscarBakker) [reply src/signals/incident-management/containers/IncidentDetail/components/ForwardToExternal/ForwardToExternal.tsx:185]: aah old comment

**Pull request merged**